src/components/TextInput/BaseTextInput.js

```diff
@@ -274,6 +273,7 @@ class BaseTextInput extends Component {
                                         onPressOut={this.props.onPress}
                                         showSoftInputOnFocus={!this.props.disableKeyboard}
                                         keyboardType={getSecureEntryKeyboardType(this.props.keyboardType, this.props.secureTextEntry, this.state.passwordHidden)}
+                                        value={this.props.value}
```

Comment:
Also, use of both `state.value` and `props.value` in this component is confusing.

Can we use `state.value` in all the possible places?